qiskit/providers/backend_compat.py

```diff
@@ -13,24 +13,367 @@
+        name = (spec["name"]).lower()
+        if name == "operational":
+            in_data[name] = bool(spec["value"])
+        elif name in IBMQubitProperties.__slots__:
+            in_data[name] = apply_prefix(value=spec["value"], unit=spec.get("unit", None))
+    return IBMQubitProperties(**in_data)  # type: ignore[no-untyped-call]
+
+
+def _decode_instruction_property(
+    gate_spec: GateSchema,
+):
+    """Decode gate property data to generate InstructionProperties instance.
+
+    Args:
+        gate_spec: List of gate property dictionary.
+
+    Returns:
+        An ``InstructionProperties`` instance and a boolean value representing
+        if this gate is operational.
+    """
+
+    # importing pacakges where they are needed, to avoid cyclic-import.
+    from qiskit.transpiler.target import InstructionProperties
+
+    in_data = {}
+    operational = True
+    for param in gate_spec.parameters:
+        if param.name == "gate_error":
+            in_data["error"] = param.value
+        if param.name == "gate_length":
+            in_data["duration"] = apply_prefix(value=param.value, unit=param.unit)
+        if param.name == "operational" and not param.value:
+            operational = bool(param.value)
+    return InstructionProperties(**in_data), operational
+
+
+def _decode_measure_property(qubit_specs: List[Dict]):
+    """Decode qubit property data to generate InstructionProperties instance.
+
+    Args:
+        qubit_specs: List of qubit property dictionary.
+
+    Returns:
+        An ``InstructionProperties`` instance.
+    """
+
+    # importing pacakges where they are needed, to avoid cyclic-import.
+    from qiskit.transpiler.target import InstructionProperties
+
+    in_data = {}
+    for spec in qubit_specs:
+        name = spec["name"]
+        if name == "readout_error":
+            in_data["error"] = spec["value"]
+        if name == "readout_length":
+            in_data["duration"] = apply_prefix(value=spec["value"], unit=spec.get("unit", None))
+    return InstructionProperties(**in_data)
+
+
+def qubit_props_list_from_props(
+    properties: BackendProperties,
+) -> List[QubitProperties]:
+    """Uses BackendProperties to construct
+    and return a list of QubitProperties.
+    """
+    qubit_props: List[QubitProperties] = []
+    for qubit, _ in enumerate(properties.qubits):
+        try:
+            t_1 = properties.t1(qubit)
+        except BackendPropertyError:
+            t_1 = None
+        try:
+            t_2 = properties.t2(qubit)
+        except BackendPropertyError:
+            t_2 = None
+        try:
+            frequency = properties.frequency(qubit)
+        except BackendPropertyError:
+            frequency = None
+        qubit_props.append(
+            QubitProperties(  # type: ignore[no-untyped-call]
+                t1=t_1,
+                t2=t_2,
+                frequency=frequency,
+            )
+        )
+    return qubit_props
+
+
+def convert_to_target_legacy(
```

Comment:
We would like to replace this `convert_to_target_legacy` with new one.
Maybe you separate them because these have two different inputs.
I think that it might be better to deprecate unnecessary inputs in new `convert_to_target`. 
In Qiskit development process, we need to add `deprecate_arg` before change API. (you can see some examples if you search `@deprecate_arg` in qiskit.)

Reply:
I have not deprecated any argument of `convert_to_target` because these arguments are used in `BackendV2Converter` of `backend_compat.py`.
Aer specifically uses argument `custom_name_mapping` to pass in all aer related instructions to `target`. The reason I seperated old codes into `convert_to_target_legacy` was because in this PR I just focused on introducing `convert_to_target` to `FakeBackendV2`.
Now, there is no `convert_to_target_legacy`, only one function `convert_to_target` is passed to both `FakeBackendV2` and `BackendV2Converter`.
All arguments of `convert_to_target` are relevant and works fine to my knowledge :)